packages/floating-vue/nuxt.mjs

```diff
@@ -1,8 +1,48 @@
-export default async function (_, _nuxt) {
+import {
+  FloatingVueDirectives,
+  enableDirectives,
+  prepareFloatingVueComponents,
+} from 'floating-vue/unimport-presets'
+
+/**
+ * Nuxt module implementation.
+ * @param options {import('./nuxt').FloatingVueNuxtOptions}
+ * @param _nuxt {import('@nuxt/schema').Nuxt}
+ * @returns {Promise<void>}
+ */
+export default async function (options = {}, _nuxt) {
   const { addPluginTemplate } = await import('@nuxt/kit')
 
+  /** @type {import('@nuxt/schema').Nuxt} */
   const nuxt = (this && this.nuxt) || _nuxt
 
+  const imports = nuxt.options.imports
+  imports.addons = enableDirectives(imports.addons)
+
+  nuxt.hook('imports:sources', (sources) => {
+    sources.push(
+      FloatingVueDirectives(options.directives),
+    )
+  })
+
+  nuxt.hook('components:extend', async (registry) => {
```

Comment:
Remove async

Reply:
I guess we don't need the plugin using Nuxt auto-import stuff: just register th eplugin if imports is disabled or add a new option to the module to use auto-import or global